model/src/test/kotlin/spdx/SpdxExpressionLexerTest.kt

```diff
@@ -0,0 +1,89 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.spdx
+
+import io.kotlintest.shouldBe
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.WordSpec
+
+import org.antlr.v4.runtime.BaseErrorListener
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.RecognitionException
+import org.antlr.v4.runtime.Recognizer
+
+class SpdxExpressionLexerTest : WordSpec() {
+    init {
+        "SpdxExpressionLexer" should {
+            "create the correct tokens for a valid expression" {
+                val testExpression = "(license-1 AND OR license.2 WITH 0LiCeNsE-3+) and (or with license-4 license-.5(("
+
+                val charStream = CharStreams.fromString(testExpression)
+                val lexer = SpdxExpressionLexer(charStream)
+                val tokens = lexer.allTokens.map { it.type to it.text }
+
+                tokens shouldBe listOf(
+                        SpdxExpressionLexer.OPEN to "(",
+                        SpdxExpressionLexer.IDSTRING to "license-1",
+                        SpdxExpressionLexer.AND to "AND",
+                        SpdxExpressionLexer.OR to "OR",
+                        SpdxExpressionLexer.IDSTRING to "license.2",
+                        SpdxExpressionLexer.WITH to "WITH",
+                        SpdxExpressionLexer.IDSTRING to "0LiCeNsE-3",
+                        SpdxExpressionLexer.PLUS to "+",
+                        SpdxExpressionLexer.CLOSE to ")",
+                        SpdxExpressionLexer.AND to "and",
+                        SpdxExpressionLexer.OPEN to "(",
+                        SpdxExpressionLexer.OR to "or",
+                        SpdxExpressionLexer.WITH to "with",
+                        SpdxExpressionLexer.IDSTRING to "license-4",
+                        SpdxExpressionLexer.IDSTRING to "license-.5",
+                        SpdxExpressionLexer.OPEN to "(",
+                        SpdxExpressionLexer.OPEN to "("
+                )
+            }
+
+            "fail for an invalid expression" {
+                val testExpression = "/"
+
+                val charStream = CharStreams.fromString(testExpression)
+                val lexer = SpdxExpressionLexer(charStream)
+                lexer.removeErrorListeners()
+                lexer.addErrorListener(object : BaseErrorListener() {
+                    override fun syntaxError(
+                            recognizer: Recognizer<*, *>?,
+                            offendingSymbol: Any?,
+                            line: Int,
+                            charPositionInLine: Int,
+                            msg: String?,
+                            e: RecognitionException?
+                    ) {
+                        throw Error(msg)
+                    }
+                })
```

Comment:
Above I asked to extract a function to remove code redundancy. 
Seeing this I think you could extract lines 65 through 79 into a function:

fun getTokensForExpression(expression: String)

which would result in quite compact test cases.

Reply:
Done.